odoo/fields.py

```diff
@@ -970,8 +960,17 @@ def is_accessible(self, env):
     # Conversion of values
     #
 
-    def convert_to_column(self, value, record, values=None, validate=True):
-        """ Convert ``value`` from the ``write`` format to the SQL format. """
+    def convert_to_column(self, value, record, values=None, validate=True, operation='create'):
```

Comment:
I think it is a good idea to split this function is multiple ones instead of having a default `operation` argument.
Even in the comment the abstraction of sub-classes is leaking here.

Reply:
Do you have any idea how we name them?
convert_to_column_flush
convert_to_column_create
convert_to_column_search
?

Reply:
What do you think about?

convert_to_database_insert
convert_to_database_update
convert_to_database_select